ksql-engine/src/test/java/io/confluent/ksql/analyzer/AnalyzerTest.java

```diff
@@ -238,4 +255,181 @@ public void shouldCreateCorrectSinkKsqlTopic() {
+            .collect(Collectors.toList());
+    final CreateStreamAsSelect createStreamAsSelect = (CreateStreamAsSelect) statements.get(0);
+    final Table intoTable = new Table(QualifiedName.of(createStreamAsSelect.getName().toString()));
+    intoTable.setProperties(createStreamAsSelect.getProperties());
+    final QuerySpecification querySpecification = (QuerySpecification) createStreamAsSelect.getQuery().getQueryBody();
+    final QuerySpecification newQuerySpecification = new QuerySpecification(
+            querySpecification.getSelect(),
+            intoTable,
+            true,
+            querySpecification.getFrom(),
+            querySpecification.getWindowExpression(),
+            querySpecification.getWhere(),
+            querySpecification.getGroupBy(),
+            querySpecification.getHaving(),
+            querySpecification.getLimit()
+    );
+    final Analysis analysis = new Analysis();
+    final Analyzer analyzer = new Analyzer("sqlExpression", analysis, jsonMetaStore, "");
+    analyzer.visitQuerySpecification(newQuerySpecification, new AnalysisContext(null));
+
+    assertThat(analysis.getIntoProperties().get(DdlConfig.VALUE_AVRO_SCHEMA_FULL_NAME).toString(), equalTo("'com.custom.schema'"));
+  }
+
+  @Test
+  public void shouldUseImplicitNamespaceForAvroSchema() {
+    final String simpleQuery = "CREATE STREAM FOO WITH (VALUE_FORMAT='AVRO', KAFKA_TOPIC='TEST_TOPIC1') AS SELECT col0, col2, col3 FROM test1 WHERE col0 > 100;";
+    final List<Statement> statements = getPreparedStatements(simpleQuery, jsonMetaStore)
+            .stream()
+            .map(PreparedStatement::getStatement)
+            .collect(Collectors.toList());
+    final CreateStreamAsSelect createStreamAsSelect = (CreateStreamAsSelect) statements.get(0);
+    final Table intoTable = new Table(QualifiedName.of(createStreamAsSelect.getName().toString()));
+    intoTable.setProperties(createStreamAsSelect.getProperties());
+    final QuerySpecification querySpecification = (QuerySpecification) createStreamAsSelect.getQuery().getQueryBody();
+    final QuerySpecification newQuerySpecification = new QuerySpecification(
+            querySpecification.getSelect(),
+            intoTable,
+            true,
+            querySpecification.getFrom(),
+            querySpecification.getWindowExpression(),
+            querySpecification.getWhere(),
+            querySpecification.getGroupBy(),
+            querySpecification.getHaving(),
+            querySpecification.getLimit()
+    );
+    final Analysis analysis = new Analysis();
+    final Analyzer analyzer = new Analyzer("sqlExpression", analysis, jsonMetaStore, "");
+    analyzer.visitQuerySpecification(newQuerySpecification, new AnalysisContext(null));
+
+    assertThat(analysis.getIntoProperties().get(DdlConfig.VALUE_AVRO_SCHEMA_FULL_NAME).toString(), equalTo(KsqlConstants.DEFAULT_AVRO_SCHEMA_FULL_NAME));
+  }
+
+  @Test
+  public void shouldUseExplicitNamespaceWhenFormatIsInheritedForAvro() {
```

Comment:
What about a test around inheriting the namespace itself for `CREATE STREAM AS` and `CREATE TABLE AS` statements? Isn't this desirable behavior?

Reply:
I would actually prefer to have the default always revert to `DEFAULT_AVRO_SCHEMA_FULL_NAME`.

Since the new property sets both the namespace and the name of the schema, I wouldn't like to see two different schemas with the same name (although seeing them with the same namespace could make sense).

Thoughts?

Reply:
Wait, I see what you mean now. I'll add that test.

Reply:
@SamirHafez one reason to inherit the namespace is that we already inherit the value type and also other properties (like number of partitions). I guess you are a target user, so what would you like to see? IF you are setting custom namespaces, would you want some outputs to default to the original?

Reply:
cc @confluentinc/ksql , more opinions welcome.

Reply:
I wonder if we should be able to set the default namespace via a property? Then i think we could just stick with the way it is, i.e., use the default unless it is specified. Also, is it true that if we don't specify a `VALUE_AVRO_FULL_SCHEMA_NAME` that the schema name is identical for every query? If so that seems pretty bad ( I know that is not part of this PR). If it is then i think we should probably have a follow up PR that changes it to use the Table/Stream name or similar.

Reply:
@dguy @big-andy-coates had suggested to allow a configurable default namespace back when I started this PR. We decided to keep that logic out of this PR and do it in a subsequent one.

Reply:
@apurvam As a user I would explicitly set the namespace for my schemas, and if I didn't set it, I would expect it to be the default.

Reply:
> Also, is it true that if we don't specify a VALUE_AVRO_FULL_SCHEMA_NAME that the schema name is identical for every query? 

It looks that way. I think enabling a user to set the default namespace is a good followup. However, that will still leave the possibility that all schemas have the same name. 

One of the risks I can see of changing the schema name inheritance behavior is upgrade compatibility: if we allow people to change the default schema namespace, then after an upgrade KSQL will probably try to register a second schema for the same topic, which probably will cause issues. 

I think for now, sticking with explicit naming makes sense. I'm happy with merging this if we can have a test case which ensures that we _don't_ inherit schema names and namespaces in CSAS / CTAS statements, @SamirHafez .